### ISSUE TITLE
LanguageToggle handle locale prefix incorrectly

### ISSUE THREAD

**Issue by Yinzo** [fuma-nama/fumadocs#1201]
### To Reproduce

1. Set up i18n (I used fumadocs-cli)
2. Set at least 2 languages, with hideLocale set to `never`
3. Use LanguageToggle to switch languages

Issue:
When switching languages, an extra locale prefix is added to the pathname, resulting in URLs like `/cn/en/docs`.
This issue affects the hideLocale `default-locale` setting and switching from non-default locales to other locales.

### Current vs. Expected behavior

Current behavior: 
Navigating to an incorrect pathname with an extra locale prefix, e.g., `/cn/en/docs`

Expected behavior: 
Replace the locale segment if it is a locale prefix; otherwise, insert it at the beginning of the path.

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.1.0: Thu Oct 10 21:06:23 PDT 2024; root:xnu-11215.41.3~3/RELEASE_ARM64_T8132
  Available memory (MB): 32768
  Available CPU cores: 10
Binaries:
  Node: 20.18.1
  npm: 10.8.2
  Yarn: 1.22.22
  pnpm: N/A
Relevant Packages:
  next: 15.1.3 // Latest available version is detected (15.1.3).
  eslint-config-next: 15.1.2
  react: 19.0.0
  react-dom: 19.0.0
  typescript: 5.7.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

UI

### Additional context

![Snipaste_2024-12-29_01-58-36](https://github.com/user-attachments/assets/a700c01e-6abd-4ad3-a14e-87997989d2b6)

I added some logs to i18n.js, and we can see that the `segment after ` has two locales. The given locale parameter is `cn` and the URL is `/en`. 

https://github.com/fuma-nama/fumadocs/blob/61f0e3ad7628463a7d21face0babce892f814796/packages/ui/src/i18n.tsx#L46-L58

**Comment by fuma-nama** [comment]
You probably passed invalid `locales` to `I18nProvider`

**Comment by ariesclark** [comment]
I encounter this as well, ``locale`` is certainly valid.

**Comment by fuma-nama** [comment]
nvm it's a mistake

**Comment by ariesclark** [comment]
confirmed fix in e95be52, thanks for the fast patch.
